corehq/apps/userreports/sql/adapter.py

```diff
@@ -260,6 +260,87 @@ def doc_exists(self, doc):
             return session.query(query.exists()).scalar()
 
 
+class MultiDBSqlAdapter(object):
+
+    mirror_adapter_cls = IndicatorSqlAdapter
+
+    def __init__(self, config, override_table_name=None):
+        config.validate_db_config()
+        self.config = config
+        self.main_adapter = self.mirror_adapter_cls(config, override_table_name)
+        self.all_adapters = []
+        engine_ids = self.config.mirrored_engine_ids + \
+            [self.main_adapter.engine_id]  # include the main primary adapter
```

Comment:
Adding this here means you now have 2 adapter instances for this engine ID. You could just add `self.main_adapter` to `self.all_adapters`. If there's a concern that the main engine_id will point to the same DB as one of the mirrored engine IDs then shouldn't that be included in validation since it seems likely to be an error.

Reply:
https://github.com/dimagi/commcare-hq/pull/24046/commits/537f972db4f0157f75d0cd0d36cce10433ae5f60